### ISSUE TITLE
Is it possible to ignore the version of a dependency and use it directly if present in VSCode ?

### ISSUE THREAD

**Issue by fmarot** [microsoft/vscode-maven#893]
## Suggestion

Hello I do not know if it's already possible in VSCode so I ask here.
Imagine I have 2 Maven projects in the IDE:
aaa in v1
bbb in v1

if aaa references bbb v1 in its pom, it's ok: a code update in bbb will be 'seen' when executing aaa.
Now, I would like code updates in bbb to be also 'seen' by aaa even if aaa reference bbb V1 while bbb being **V2** in VSCode.

Is is possible ?

## Use Cases

My use case is related to having multiple git repositories each having projects in different versions. I would like to avoid having to change the poms (in fact, the grand scheme is related to CIFriendly behavior but I prefer to focus on the simple subject mentioned abvove)

## Examples

![image](https://user-images.githubusercontent.com/633019/198843388-94560ad2-0baa-424a-9e6c-7448198baecf.png)

**Comment by Eskibear** [comment]
> Is is possible ?

No. There would be ambiguity. How does the extension know whether you want to update bbb's version in project `aaa` or not?


> I would like to avoid having to change the poms

`dependencyManagement` is designed for it, where you can manage dependency versions in a single place.

**Comment by fmarot** [comment]
Hello Eskibear, thanks for the reply
> No. There would be ambiguity. How does the extension know whether you want to update bbb's version in project aaa or not?

the extension would not know. It would be a on/off setting. If 'on', the rationale would be "if I have added a folder to the workspace, it's because I WANT to use the code in this folder, whatever the version. So just ignore the version mismatch if the option is 'on' and use this code.

> dependencyManagement is designed to allow to avoid changing the code

Not really, changing dependencyManagement IS changing the pom. In my view, the real version, including the gitbranch as a suffix, is deterrmined by the CI and dynamically passed. The flaten plugin then will deploy poms containing this real version to the shared repository. I'd like the devs working on 2 different git repos to be able to work simultaneously on both codebase, to be able to debug directly from within the IDE even if the versions are not the same because he knows in the end the versions will be the same when set by the CI.

In my naive (!) opinion, it would not be really difficult to ignore the version when resolving. It would save so much time by making Maven both CIFriendly AND debuggable n the IDE.

**Comment by Eskibear** [comment]
> to ignore the version when resolving. It would save so much time by making Maven both CIFriendly AND debuggable n the IDE.

I don't think it's a common use case given my knowledge of Maven. In my opinion, for CI pipelines you can have scripts to set corresponding "real version", and so for your dev environment. 
For your use case, AFAIK in Maven's spec you cannot define a `relativePath` for a `dependency` node, but maybe you can make its `scope` as "provided" and provide the `systemPath` pointing to bbb's artifact path, which is relatively stable.

**Comment by fmarot** [comment]
The fact is I do not want to change the pom files. Nor any other file.

It clearly not a common use-case for little self-sufficient Maven Project, but as soon as you start work accross multiple git repositories each contaning multi-modules Maven projects you feel the pain.

Technically, would it be difficult to implement ?
